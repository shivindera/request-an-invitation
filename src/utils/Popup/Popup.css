.broccoli-popup-mask {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1;
  background: rgb(0 0 0 / 50%);
}

.broccoli-popup {
  min-width: 480px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background: white;
  border: 1px solid gray;
  border-radius: 4px;
  z-index: 2;
  padding: 16px 24px;
}

.broccoli-popup-close {
  position: absolute;
  width: 20px;
  height: 20px;
  top: 8px;
  right: 8px;
  cursor: pointer;
  z-index: 3;
}

.broccoli-popup-close::before,
.broccoli-popup-close::after {
  content: '';
  position: absolute;
  top: 8px;
  left: -2px;
  right: 0;
  height: 2px;
  background: #000;
  border-radius: 4px;
}

.broccoli-popup-close::before {
  transform: rotate(45deg);
}

.broccoli-popup-close::after {
  transform: rotate(-45deg);
}

@media only screen and (max-width: 640px) {
  .broccoli-popup {
    max-width: 360px;
    min-width: auto;
    width: 100%;
  }
}
