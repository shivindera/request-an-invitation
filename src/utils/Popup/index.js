// Popup component to provide popup functionality

import './Popup.css';

const Popup = (props) => {
  // isOpen - Flag to show or hide the popup
  // onClose - Callback to when the popup is closed or a click is made outside the popup
  const { children, isOpen, onClose } = props;
  return isOpen ? (
    <>
      <div className='broccoli-popup-mask' onClick={() => onClose()}></div>
      <div className='broccoli-popup'>
        <span className='broccoli-popup-close' onClick={() => onClose()} />
        <div className='broccoli-popup-container'>{children}</div>
      </div>
    </>
  ) : null;
};

export default Popup;
