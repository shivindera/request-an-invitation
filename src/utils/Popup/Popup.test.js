import { fireEvent, render } from '@testing-library/react';
import Popup from './';

describe('Test the Button functionality', () => {
  const baseProps = {
    isOpen: false,
    onClose: jest.fn(),
  };

  it('shows nothing when isOpen is false', () => {
    const input = render(<Popup {...baseProps} />);
    expect(input.container).toMatchSnapshot();
  });

  it('shows popup skeleton with data when isOpen is true', () => {
    const input = render(
      <Popup {...baseProps} isOpen={true}>
        Test Popup
      </Popup>
    );
    expect(input.container).toMatchSnapshot();
  });

  it('should trigger onClose when the close button is clicked', () => {
    const input = render(<Popup {...baseProps} isOpen={true} />);
    expect(input.container.getElementsByClassName('broccoli-popup-close').length).toBe(1);
    fireEvent.click(input.container.getElementsByClassName('broccoli-popup-close')[0]);
    expect(baseProps.onClose).toHaveBeenCalledTimes(1);
  });

  it('should trigger onClose a click is done outside the popup', () => {
    const input = render(<Popup {...baseProps} isOpen={true} />);
    expect(input.container.getElementsByClassName('broccoli-popup-mask').length).toBe(1);
    fireEvent.click(input.container.getElementsByClassName('broccoli-popup-mask')[0]);
    expect(baseProps.onClose).toHaveBeenCalledTimes(1);
  });
});
