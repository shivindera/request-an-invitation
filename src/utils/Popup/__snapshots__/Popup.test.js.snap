// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Button functionality shows nothing when isOpen is false 1`] = `<div />`;

exports[`Test the Button functionality shows popup skeleton with data when isOpen is true 1`] = `
<div>
  <div
    class="broccoli-popup-mask"
  />
  <div
    class="broccoli-popup"
  >
    <span
      class="broccoli-popup-close"
    />
    <div
      class="broccoli-popup-container"
    >
      Test Popup
    </div>
  </div>
</div>
`;
